### PR TITLE
NEO-2348 rename

### PR DESCRIPTION
**Pull request by yangAtSpoken**
[Link to updated components in Deploy Preview](https://deploy-preview-532--neo-react-library-storybook.netlify.app/?path=/story/components-table--advanced-filtering-and-sorting)

**Before tagging the team for a review, I have done the following:**

- [x] run `yarn all` locally: ensures that all tests pass, formatting is done, types pass, and builds pass
- [x] reviewed my code changes
- [x] updated the Deploy Preview link at the top of this PR (or remove it if not applicable)
- [ ] updated the Figma referense link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [ ] done an accessibility check on my work (tested with Chrome's `axe Dev Tools`, Mac's VoiceOver, etc.)
- [ ] tagged `@avaya-dux/dux-design` if any visual changes have occurred
- [x] tagged `@avaya-dux/dux-devs`

`@avaya-dux/dux-devs`: mostly refactor and remove column filter UI from header cell

- rename allowColumnFilter to allowToggleColumnVisibility
- allow custom column filter in the column drawer
- allow Select to have an SelectOption and an SelectOption[] as children
Screenshot 2024-09-13 at 11 07 53 AM

## Summary by CodeRabbit

- **New Features**
	- Enhanced handling of child elements in the Select component for improved usability.
	- Introduced a new functionality for toggling column visibility in the Table component.

- **Bug Fixes**
	- Improved type safety in the Select component, reducing potential runtime errors.

- **Refactor**
	- Simplified filtering logic and updated properties in the Table component to focus on column visibility.
	- Streamlined rendering logic in the TableHeader and TableToolbar components.

- **Documentation**
	- Updated type definitions to reflect changes in component properties and enhance clarity.

### CODE REVIEW

File: src/components/Table/Table.stories.tsx [hunk 1]
```diff
@@ -258,7 +257,7 @@ export const ServerSidePagination = () => {
 				handlePageChange={fetchData}
 				handleSearch={searchDebounced}
 				itemsPerPageOptions={[5, 10, 20, 50]}
-				allowColumnFilter
+				allowToggleColumnVisibility
```
Comment (joe-s-avaya) [inline src/components/Table/Table.stories.tsx:260]: 👍

File: src/components/Table/Table.stories.tsx [hunk 2]
```diff
@@ -360,29 +358,44 @@ export const AdvancedFilteringAndSorting = () => {
 					return Array.from(optionSet.values());
 				}, [id, preFilteredRows]);
 
+				const all = useMemo(
+					() => (
+						<SelectOption key="All" value="">
+							All
```
Comment (joe-s-avaya) [inline src/components/Table/Table.stories.tsx:364]: must be translated

File: src/components/Table/Table.stories.tsx [hunk 3]
```diff
@@ -360,29 +358,44 @@ export const AdvancedFilteringAndSorting = () => {
...
+						<SelectOption key={option as string} value={option as string}>
+							{(option as string).toUpperCase()}
+						</SelectOption>
+					));
+				}, [options]);
+
 				return (
 					<div style={{ margin: "0px 0px -8px 0px" }}>
-						<SelectNative
+						<Select
```
Comment (joe-s-avaya) [inline src/components/Table/Table.stories.tsx:380]: Oh good! I'm glad that you've updated to use our own `Select`, and that it's working properly!

File: src/components/Table/Table.stories.tsx [hunk 4]
```diff
@@ -360,29 +358,44 @@ export const AdvancedFilteringAndSorting = () => {
 					return Array.from(optionSet.values());
 				}, [id, preFilteredRows]);
 
+				const all = useMemo(
+					() => (
+						<SelectOption key="All" value="">
```
Comment (joe-s-avaya) [inline src/components/Table/Table.stories.tsx:363]: Shouldn't the `value` be `"all"`?

File: src/components/Select/utils/SelectTypes.tsx [hunk 5]
```diff
@@ -18,7 +18,8 @@ type LabelOrAriaLabelProps =
 export type SelectProps = {
 	children?:
 		| ReactElement<SelectOptionProps>
-		| ReactElement<SelectOptionProps>[];
+		| ReactElement<SelectOptionProps>[]
+		| (ReactElement<SelectOptionProps> | ReactElement<SelectOptionProps>[])[];
```
[the remark this answers is not in the document]
Comment (joe-s-avaya) [reply src/components/Select/utils/SelectTypes.tsx:22]: oh wait, I missed that it was wrapped in parentheses, nvm, resolving

File: src/components/Table/Table.stories.tsx (hunk 2 above)
Comment (joe-s-avaya) [reply src/components/Table/Table.stories.tsx:364]: oh wait, this is a Story, nvm, resolving

File: src/components/Table/Table.stories.tsx (hunk 4 above)
Comment (joe-s-avaya) [reply src/components/Table/Table.stories.tsx:363]: Based on line 386 (`onFilterValueChange(value === "All" ? "" : value);`)

**Review by joe-s-avaya — APPROVED** [review]
A few questions but nothing I want to block on.

**Review by enrique-prado — APPROVED** [review]

File: src/components/Table/Table.stories.tsx (hunk 4 above)
Comment (yangAtSpoken) [reply src/components/Table/Table.stories.tsx:363]: Select automatically sets value to body when it is empty.  But I agree setting it to "All" explicitly is better.

File: src/components/Select/Select.tsx [hunk 6]
```diff
@@ -100,18 +107,20 @@ export const Select = (props: SelectProps) => {
...
+
+		return Children.map(childrenArray, (child) => {
+			const childprops = { ...child.props };
+			if (!childprops.value) {
+				childprops.value = childprops.children;
+			}
+
+			return childprops;
+		});
+	}, [children]);
```
[the remark this answers is not in the document]
Comment (yangAtSpoken) [reply src/components/Select/Select.tsx:123]: children prop could be any of the 3 cases: a single element, an array, and a mix of singles and arrays.  In the story, children is a single element (the All option ) plus an array (set of options returned from the map).  toArray handles all these 3 use cases.   In the story, if another column filter returns empty rows, this Select will end up with a single option.

[the remark this answers is not in the document]
Comment (joe-s-avaya) [reply src/components/Select/Select.tsx:123]: ah, makes sense, thanks for the explanation

**Comment by enrique-prado** [comment]
> @enrique-prado is the Drawer supposed to be constrained to the table height? When there is only a single rowed displayed it looks pretty bad.
> 
> Screenshot 2024-09-13 at 10 53 04

It's not supposed to be constrained to the table's height.  At least that's my understanding. Ideally it would be as high as the outer container (the view or section that contains the table)

**Pull request merged**